### ISSUE TITLE
RightCurly: handling of try-with-resources on several lines with CS 7.5

### ISSUE THREAD

**Issue by don-vip** [checkstyle/checkstyle#3776]
/var/tmp $ javac Test.java

/var/tmp $ cat Test.java

```java
// License: GPL. For details, see LICENSE file.
import java.io.BufferedReader;
import java.io.IOException;
class Test {

    void test() {
        try (BufferedReader br1 = new BufferedReader(null);
             BufferedReader br2 = new BufferedReader(br1)) {
            System.out.println("test");
        } catch (IOException e) {
            System.err.println(e);
        }
    }
}
```

/var/tmp $ cat config.xml

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="RightCurly"/>
    </module>
</module>
```

/var/tmp $ java -jar checkstyle-7.5-all.jar -c config.xml Test.java

> Starting audit...
> [WARN] C:\SVN\Test.java:8:58: '}' at column 58 should be on a new line. [RightCurly]
> Audit done.

---------------

Checkstyle 7.4 did not raise error, and that's the behaviour we want.
http://checkstyle.sourceforge.net/config_blocks.html#RightCurly

**Comment by romani** [comment]
regression is confirmed, no violations on 7.4 .

looks like we lost smth at recent updates:
https://github.com/checkstyle/checkstyle/commit/4cda2565674ea4260547e663e9660df0029709d1
https://github.com/checkstyle/checkstyle/commit/2c81de1bb030dbaf3885a17cdfff8a260a7b337e

**Comment by romani** [comment]
@don-vip , if you run checkstyle on open source project please share link to your code, we will add your project to our regression testing.

**Comment by asashour** [comment]
HtmlUnit is an open-source project, and faces the same issue.

Code in https://sourceforge.net/p/htmlunit/code/HEAD/tree/trunk/htmlunit/

**Comment by don-vip** [comment]
Thank you! The project is there: https://josm.openstreetmap.de/svn/trunk/

The classes that provoque the error:

```xml
<file name="C:\SVN\josm\core\src\org\openstreetmap\josm\io\BoundingBoxDownloader.java">
<error line="169" column="72" severity="warning" message="&apos;}&apos; à la colonne 72 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
<error line="176" column="72" severity="warning" message="&apos;}&apos; à la colonne 72 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
<error line="188" column="72" severity="warning" message="&apos;}&apos; à la colonne 72 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\src\org\openstreetmap\josm\io\NoteExporter.java">
<error line="42" column="56" severity="warning" message="&apos;}&apos; à la colonne 56 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\src\org\openstreetmap\josm\io\imagery\ImageryReader.java">
<error line="86" column="40" severity="warning" message="&apos;}&apos; à la colonne 40 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\functional\org\openstreetmap\josm\data\osm\TaginfoTestIT.java">
<error line="55" column="55" severity="warning" message="&apos;}&apos; à la colonne 55 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\functional\org\openstreetmap\josm\io\MultiFetchServerObjectReaderTest.java">
<error line="183" column="10" severity="warning" message="&apos;}&apos; à la colonne 10 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\functional\org\openstreetmap\josm\io\OsmServerBackreferenceReaderTest.java">
<error line="195" column="10" severity="warning" message="&apos;}&apos; à la colonne 10 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\unit\org\openstreetmap\josm\data\AutosaveTaskTest.java">
<error line="193" column="113" severity="warning" message="&apos;}&apos; à la colonne 113 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
<error line="237" column="113" severity="warning" message="&apos;}&apos; à la colonne 113 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\unit\org\openstreetmap\josm\data\osm\NodeDataTest.java">
<error line="24" column="68" severity="warning" message="&apos;}&apos; à la colonne 68 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\unit\org\openstreetmap\josm\data\osm\WayDataTest.java">
<error line="27" column="68" severity="warning" message="&apos;}&apos; à la colonne 68 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\unit\org\openstreetmap\josm\data\projection\ProjectionRefTest.java">
<error line="101" column="83" severity="warning" message="&apos;}&apos; à la colonne 83 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
<error line="273" column="84" severity="warning" message="&apos;}&apos; à la colonne 84 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\unit\org\openstreetmap\josm\data\projection\ProjectionRegressionTest.java">
<error line="91" column="85" severity="warning" message="&apos;}&apos; à la colonne 85 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
<error line="121" column="41" severity="warning" message="&apos;}&apos; à la colonne 41 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
<file name="C:\SVN\josm\core\test\unit\org\openstreetmap\josm\data\validation\routines\DomainValidatorTestIT.java">
<error line="276" column="64" severity="warning" message="&apos;}&apos; à la colonne 64 devrait être sur une nouvelle ligne." source="com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck"/>
</file>
```

**Comment by romani** [comment]
@asashour ,

At what version of your code have no violation ? trunk and latest release do have problems.
I can use your repo only if code has no validation errors.
```
$ svn export https://svn.code.sf.net/p/htmlunit/code/tags/HtmlUnit-2.24 htmlunit
..
$ cd htmlunit3/
$mvn checkstyle:check
...
[INFO] Starting audit...
[ERROR] /home/rivanov/nfs/temp/4/htmlunit3/src/test/java/com/gargoylesoftware/htmlunit/source/SVN.java:0: Missing package-info.java file. [JavadocPackage]
[ERROR] /home/rivanov/nfs/temp/4/htmlunit3/src/test/java/com/gargoylesoftware/htmlunit/gae/GAETestRunner.java:0: Missing package-info.java file. [JavadocPackage]
Audit done.
```

**Comment by asashour** [comment]
Hi Ronam,

Thanks for looking into this.

I have to disagree, we use Continuous Integration, and `checkstyle:checkstyle` is part of the each and every build, it won't be green otherwise.

I believe you have local customization that fails HtmlUnit build, both locally on developers machines and on the build server, checkstyle passes.

Anyhow, I just committed the two `package-info.java` missing in the test package, hope it passes for you now, please let me know if not.

You can see the result of the build in [here](https://ci.canoo.com/teamcity/viewLog.html?buildTypeId=HtmlUnit_FastBuild&buildId=lastSuccessful&tab=artifacts) (login as `guest`).

Thanks a lot

**Comment by romani** [comment]
@don-vip , I just added your project to our regression testing and your trunk pass on latest code
https://travis-ci.org/checkstyle/checkstyle/jobs/197120534#L9456

PR: https://github.com/checkstyle/checkstyle/pull/3783
There are no code changes from 7.5 release - https://github.com/checkstyle/checkstyle/commits/master

Did I miss smth in my commands to run checkstyle on your project ?

**Comment by romani** [comment]
@asashour , thanks for a fix, now it works.
please do a favor and move "7.4" version to properties of pom.xml this will allow me to change version from command line to simplify build execution on our side.

**Comment by asashour** [comment]
Done, thanks.

**Comment by don-vip** [comment]
@romani "ant checkstyle" produces an output file called "checkstyle-josm.xml", is it parsed by your script or does it assume we print the errors on standard output? Also do you want us to remove the version number as well?

**Comment by romani** [comment]
@don-vip , just let me know how can I fail the build.
Do you have CI for checkstyle ?  How you control that there are no checkstyle violations in build ?

> Also do you want us to remove the version number as well?

if you update your build.xml some how to make me change checkstyle version from command line it will be very helpful or any other simplification to avoid my changes of your sources.

**Comment by don-vip** [comment]
@romani yes our Jenkins is configured to report Checkstyle violations, yo can see the trend here: https://josm.openstreetmap.de/jenkins/job/JOSM/jdk=JDK8/checkstyle/trendDetails/

I just have removed the version number in https://josm.openstreetmap.de/changeset/11524/josm/

**Comment by don-vip** [comment]
@romani you have to add "failOnViolation=true" argument in our build.xml when we call checkstyle target. Depending on if you want the violations on standard outputs or not, you will probably have to change the formattre as well.

**Comment by romani** [comment]
@don-vip , it is better to do less changes to your code, I will parse output file.

But unfortunately Ia can not make Travis to analyze your repo, Very weird  error `2563 Killed                  ant -v checkstyle`
https://travis-ci.org/checkstyle/checkstyle/jobs/197385378#L418, and it works on my local .
I did not use ANT foe long time... , I do not what to do with this.

**Comment by romani** [comment]
openstreetmap integration to CI is moved to separate issue - #3786

**Comment by romani** [comment]
fix is merged.
